src/math/nq.rs

```diff
@@ -98,353 +69,5 @@ type Color = Quad<i32>;
-                self.colormap[smallpos] = q;
-            }
-            // smallval entry is now in position i
-            if smallval != previouscol {
-                self.netindex[previouscol] = (startpos + i) >> 1;
-                for j in (previouscol + 1)..smallval {
-                    self.netindex[j] = i
-                }
-                previouscol = smallval;
-                startpos = i;
-            }
-        }
-        let max_netpos = self.netsize - 1;
-        self.netindex[previouscol] = (startpos + max_netpos) >> 1;
-        for j in (previouscol + 1)..256 {
-            self.netindex[j] = max_netpos
-        } // really 256
-    }
-
-    /// Search for best matching color
-    fn search_netindex(&self, b: u8, g: u8, r: u8, a: u8) -> usize {
-        let mut bestd = 1 << 30; // ~ 1_000_000
-        let mut best = 0;
-        // start at netindex[g] and work outwards
-        let mut i = self.netindex[g as usize];
-        let mut j = if i > 0 { i - 1 } else { 0 };
-
-        while (i < self.netsize) || (j > 0) {
-            if i < self.netsize {
-                let p = self.colormap[i];
-                let mut e = p.g - i32::from(g);
-                let mut dist = e * e; // index key
-                if dist >= bestd {
-                    break;
-                } else {
-                    e = p.b - i32::from(b);
-                    dist += e * e;
-                    if dist < bestd {
-                        e = p.r - i32::from(r);
-                        dist += e * e;
-                        if dist < bestd {
-                            e = p.a - i32::from(a);
-                            dist += e * e;
-                            if dist < bestd {
-                                bestd = dist;
-                                best = i;
-                            }
-                        }
-                    }
-                    i += 1;
-                }
-            }
-            if j > 0 {
-                let p = self.colormap[j];
-                let mut e = p.g - i32::from(g);
-                let mut dist = e * e; // index key
-                if dist >= bestd {
-                    break;
-                } else {
-                    e = p.b - i32::from(b);
-                    dist += e * e;
-                    if dist < bestd {
-                        e = p.r - i32::from(r);
-                        dist += e * e;
-                        if dist < bestd {
-                            e = p.a - i32::from(a);
-                            dist += e * e;
-                            if dist < bestd {
-                                bestd = dist;
-                                best = j;
-                            }
-                        }
-                    }
-                    j -= 1;
-                }
-            }
-        }
-        best
-    }
-}
+#[deprecated(note = "Use the `color_quant` crate instead")]
+pub type NeuQuant = color_quant::NeuQuant;
```

Comment:
That is a breaking change. We need to supply an empty shell and dispatch to the `color_quant` methods.

Reply:
Note that another crate might already depend on both crates and have individual trait impls for both types. This doesn't seem likely but let's rather avoid the risk.

Reply:
As long as use of deprecated funtion is not denied, this will not break? See: https://github.com/image-rs/image/pull/1339/commits/83a761a8ee9b697ff0c45bdd95c8a89ba6abc2a3

Running `cargo test` produces

```
...
test src/math/nq.rs - math::nq::NeuQuant (line 36) ... ok
```

Reply:
It won't break this crate but is nevertheless a breaking change according to SemVer and the Rust RFCs on the topic.

Reply:
I have no experience in dealing with this kind of scenario, so could you please make the edits that are needed?

Reply:
Sure, no problem.